### PR TITLE
Support ARM64 build binary and images

### PR DESCRIPTION
**Pull request by StLeoX**
### Support ARM64 build binary and images
- [ ] If this is non-trivial feature, paste the links/URLs to the design doc.
- [x] Update the documentation to include this new feature.
- [ ] Tests(including UT, IT, E2E) are added to verify the new feature.
- [ ] If it's UI related, attach the screenshots below.
- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Fixes apache/skywalking#12308 .

### CODE REVIEW

File: .github/workflows/publish-docker.yml [hunk 1]
```diff
@@ -79,6 +79,10 @@ jobs:
         run: |
           GOOS=windows GOARCH=amd64 make -C banyand banyand-server-static
           GOOS=windows GOARCH=amd64 make -C bydbctl build
+      - name: Build OSX binaries
+        run: |
+          GOOS=darwin GOARCH=arm64 make -C banyand banyand-server-static
+          GOOS=darwin GOARCH=arm64 make -C bydbctl build
```
Comment (wu-sheng) [inline .github/workflows/publish-docker.yml:85]: Is this AARM or ARM? The CI name seems a little tricky.

**Comment by StLeoX** [comment]
Currently, I just add `darwin-arm64` for  docker  publishing.
  
I wonder if we really need something like `banyand-server-static-darwin-arm64` in the tar release. And the binaries list  may  be semilar  with `bydnctl` ?

**Comment by wu-sheng** [comment]
> Currently, I just add `darwin-arm64` for  docker  publishing.
> 
>   
> 
> I wonder if we really need something like `banyand-server-static-darwin-arm64` in the tar release. And the binaries list  may  be semilar  with `bydnctl` ?

darwin(aarm), we don't need this for official release. But we need Linux ARM as that, same for ctl.

The issue is referring these two things.

**Comment by lujiajing1126** [comment]
Shall we consider [goreleaser](https://goreleaser.com/) to build binanries and docker images?

**Comment by wu-sheng** [comment]
We could consider this in a separate thread?

**Comment by wu-sheng** [comment]
@hanahmily Please confirm whether this is the scope of this PR. Because in the end, we will need 3 images(linux intel, linux and, dawin) in GitHub image repo and first two in the DockerHub.

**Comment by hanahmily** [comment]
The build system needs to be improved to support building multi-platform binaries, which is more complicated than I initially thought. I will address this issue. @StLeoX, I will leave you to your ospp project.

**Comment by StLeoX** [comment]
Fine

**Pull request closed without merging**